go/database/mpt/proof.go

```diff
@@ -115,3 +139,135 @@ func (p WitnessProof) AllStatesZero(root common.Hash, address common.Address, fr
 func (p WitnessProof) AllAddressesEmpty(root common.Hash, from, to common.Address) (tribool.Tribool, error) {
 	panic("not implemented")
 }
+
+// Equals returns true if the two witness proofs are equal.
+func (p WitnessProof) Equals(other WitnessProof) bool {
+	return maps.EqualFunc(p.proofDb, other.proofDb, rlpEncodedNodeEquals)
+}
+
+// MergeProofs merges the input witness proofs into the first witness proof.
+func MergeProofs(others ...WitnessProof) WitnessProof {
+	res := WitnessProof{make(proofDb)}
+	for _, other := range others {
+		res.Add(other)
+	}
+
+	return res
+}
+
+// visitWitnessPathTo visits all nodes from the input root following the input path.
+// Each encountered node is passed to the visitor.
+// If no more nodes are available on the path, the execution ends.
+// If the path does not exist, the function returns false.
+// The function returns an error if the path cannot be iterated due to error propagated from the input proof.
+// When the function reaches either an account node or a value node it is compared to the remaining input path
+// that was not iterated yet.
+// If the path matches, the function terminates and returns true.
+// It means this function can be used to find either an account node or a value node,
+// but it cannot find both at the same time.
+func visitWitnessPathTo(source proofDb, root common.Hash, path []Nibble, visitor witnessProofVisitor) (bool, error) {
+	nodeId := root
```

Comment:
nit: in this package, there is a type `NodeID` which is a different concept that what it is used for here. Maybe rename to `nodeHash`?

Reply:
fixed